Compile issues during `go get`

Go version: `go version go1.10.1 windows/amd64`
Hazelcast Go Client version: Latest commit: `0df91da3f327f8df8e0b05ead439a271484f7407`
Hazelcast server version: /
Number of the clients: /
Cluster size, i.e. the number of Hazelcast cluster members: /
OS version : Windows 10 Pro 1803

#### Expected behaviour

Successfull download and compile of the `hazelcast-go-client` package

#### Actual behaviour

The download seems to be successful, but during the compile step I get the following error:

```
# github.com/hazelcast/hazelcast-go-client/internal
..\..\hazelcast\hazelcast-go-client\internal\statistics.go:154:12: undefined: syscall.Rlimit
..\..\hazelcast\hazelcast-go-client\internal\statistics.go:155:12: undefined: syscall.Getrlimit
..\..\hazelcast\hazelcast-go-client\internal\statistics.go:155:30: undefined: syscall.RLIMIT_NOFILE
```

#### Steps to reproduce the behaviour

`go get -u "github.com/hazelcast/hazelcast-go-client"`

I also tried just downloading the package first, then cd into the package directory and then compile, but that should be the same so it also leads to the same error:
1. `go get -d "github.com/hazelcast/hazelcast-go-client"`
2. `cd $(go env GOPATH)/src/github.com/hazelcast/hazelcast-go-client`
3. `go build`

--- a/internal/statistics.go
+++ b/internal/statistics.go
@@ -26,8 +26,6 @@ import (
 
 	"time"
 
-	"syscall"
-
 	"github.com/hazelcast/hazelcast-go-client/config/property"
 	"github.com/hazelcast/hazelcast-go-client/internal/proto"
 	"github.com/hazelcast/hazelcast-go-client/internal/util/timeutil"
@@ -148,14 +146,6 @@ func (s *statistics) collectRuntimeMetrics(stats *bytes.Buffer) {
 	s.addStat(stats, "runtime.freeMemory", strconv.Itoa(int(m.HeapIdle)))
 	s.addStat(stats, "runtime.totalMemory", strconv.Itoa(int(m.HeapSys)))
 	s.addStat(stats, "runtime.usedMemory", strconv.Itoa(int(m.HeapInuse)))
-}
-
-func (s *statistics) collectOSMetrics(stats *bytes.Buffer) {
-	var limit syscall.Rlimit
-	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &limit); err == nil {
-		s.addStat(stats, "os.maxFileDescriptorCount", strconv.Itoa(int(limit.Max)))
-		s.addStat(stats, "os.openFileDescriptorCount", strconv.Itoa(int(limit.Cur)))
-	}
 }
 
 func (s *statistics) addStat(stats *bytes.Buffer, name string, value interface{}) {

--- a/internal/statistics_osmetric.go
+++ b/internal/statistics_osmetric.go
@@ -1,0 +1,31 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux darwin
+
+package internal
+
+import (
+	"bytes"
+	"strconv"
+	"syscall"
+)
+
+func (s *statistics) collectOSMetrics(stats *bytes.Buffer) {
+	var limit syscall.Rlimit
+	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &limit); err == nil {
+		s.addStat(stats, "os.maxFileDescriptorCount", strconv.Itoa(int(limit.Max)))
+		s.addStat(stats, "os.openFileDescriptorCount", strconv.Itoa(int(limit.Cur)))
+	}
+}

--- a/internal/statistics_other.go
+++ b/internal/statistics_other.go
@@ -1,0 +1,25 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !darwin
+// +build !linux
+
+package internal
+
+import (
+	"bytes"
+)
+
+func (s *statistics) collectOSMetrics(stats *bytes.Buffer) {
+}

